packages/editor/src/lib/hooks/useTLStore.ts

```diff
@@ -0,0 +1,8 @@
+import { useState } from 'react'
+import { StoreOptions, createTLStore } from '../config/createTLStore'
+
+/** @public */
+export function useTLStore(opts: StoreOptions) {
+	const [store] = useState(() => createTLStore(opts))
+	return store
+}
```

Comment:
this `useState` constant pattern is fine for top-level stuff and example where we know the inputs will never change, but it's not safe to do here. If we re-rendered this with different options, we wouldn't end up creating a new store with those options.

It's a little tricky to figure out what exactly to do here, but we could consider something like this:
```ts
const [store, setStore] = useState(() => createTLStore(opts))
const previousOpts = usePrevious(opts)
if (previousOpts !== opts) {
    const newStore = createTLStore(opts)
    setStore(newStore)
    return newStore
}
return store
```

but it does mean that if you change the inputs to a store you'll get a fresh (empty) one